### ISSUE TITLE
MoE load-balancing loss computed but not added to training loss

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#59]
# MoE load-balancing loss computed but not added to training loss

## Summary

The auxiliary load-balancing loss for the Mixture-of-Experts layer is calculated and logged but never actually incorporated into the total training loss before backpropagation. This defeats the purpose of the balancing term and may lead to expert collapse.

## Problem

The code computes the standard MoE balancing term:
```
L_aux = α · N · ∑(f_i · P_i)
```

where:
- `f_i` is the fraction of tokens routed to expert `i`
- `P_i` is the mean gate probability for expert `i`
- `N` is the number of experts
- `α` is `load_balancing_coef`

However, this term is only logged—it's not combined with the main loss before calling `backward()`. As a result, the gating network receives no gradient signal to distribute tokens evenly across experts, and may collapse to routing most tokens to just a few experts.

## Suggested Fix

Incorporate the auxiliary loss into the optimization:
```python
total_loss = main_loss + load_balancing_coef * aux_loss
total_loss.backward()
```

The `load_balancing_coef` parameter already exists but is effectively unused.

## Expected Behavior

- Auxiliary loss should contribute gradients during training
- Expert utilization should be roughly balanced (observable via logged `f_i` values)
- `load_balancing_coef` should have a measurable effect on routing behavior
